src/nvim/message.c

```diff
@@ -2383,6 +2383,15 @@ static void redir_write(char_u *str, int maxlen)
   char_u      *s = str;
   static int cur_col = 0;
 
+  // Append output to capture().
+  if (capture_ga) {
+    int len = maxlen > 0 ? maxlen : (int)STRLEN(str);
```

Comment:
Also this line is incorrect as I see handling of `maxlen` below:

1. Documentations states that you should check for `maxlen == -1`. Below code checks that `maxlen < 0`. Negation is `maxlen >= 0`. You have `maxlen > 0`.
2. Below cycle has `*s != NUL && (maxlen < 0 || (int)(s - str) < maxlen`. Note the first condition: `maxlen` is *maximal* length of the string, not *exact* length. Based on the code I see this should not lead to any bugs now, but this still is not correct. `strlen()` should be computed always. Though based on the code I could say that you may just rename `maxlen` argument of this and `msg_puts_attr_len` (the only place where `redir_write` is called not with `-1`) to just `len` and treat it as the actual length.